examples/step-32/step-32.cc

```diff
@@ -3328,9 +3328,6 @@ namespace Step32
 
         temperature_solution     = distributed_temp1;
         old_temperature_solution = distributed_temp2;
-
-        Assert(old_temperature_solution.has_ghost_elements(),
-               ExcInternalError());
```

Comment:
You could also just assert that vector has ghost elements if more than one MPI rank is used.

Reply:
True. I find the assert a bit out of place here, though (why just old_temperature? why right there, we are not using it here? etc), and wouldn't mind removing it.

Reply:
The assertion is mistaken. If you have a coarse mesh, it may be that even if you have more than one process, one process owns the entire mesh and its vectors have no ghost elements. The other processes have no cells, no ghost cells, and so store nothing and no ghost elements in their vectors either.